Oracle.md

```diff
@@ -0,0 +1,186 @@
+
+In the case where the number of digits that make up the outcome value exceeds the number of r-values that the oracle committed to, the oracle should sign the maximum possible value that he can attest to. 
+
+
+#### Example: BTC/USD rate
+
+The following example illustrates how numerical decomposition can help reduce the number of required CETs while covering a large range of possible outcomes.
+
+Alice and Bob wish to enter into a DLC at time `t` when BTC/USD ~= $10000 with a maturity time of `t+1`.
+The oracle provides 6 R-values, meaning that it can attest to outcomes in the range `[0,999999]`:
+
+```
+base: 10
+isSigned: false
+RValues: [R0, R1, R2, R3, R4, R5]
+```
+
+Alice and Bob create a DLC with "relevant" outcomes between $10000 and $10999.
+Below $10000 Alice gets everything, and above $10999 Bob gets everything.
+They use the `R0` and `R1` values to cover the cases where BTC/USD < $10000 using `R0=0` and `R1=0`.
+They use `R0`, `R1` and `R2` to cover the cases where BTC/USD > $10999 and <= $19999, using `R1=0`, `R1=1` and `R2 in [0-9]`.
+They use `R0` and `R1` to cover the cases where BTC/USD >= $20000 and < $99999, using `R0=0` and `R1 in [2,9]`.
+Finally, they use only `R0` to cover the cases where BTC/USD >= $100000 and <= $999999, using `R0 in [1-9]`.
+
+At maturity time, if the BTC/USD rate is less than $999999, the oracle signs each of the digits representing the outcome value using the digits' corresponding R value.
+If the rate is greater than $999999, the oracle signs the value $999999.
+
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`string`:`external_name`]
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`u16`:`num_outcomes`]
+   * [`u16`:`outcome_1_len`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`int32`:`start`]
+   * [`u32`:`count`]
+   * [`u16`:`step`]
+
+#### Version 0 `large_range_event_descriptor`
+
+1. type: 55306 (`large_range_event_descriptor_v0`)
+2. data:
+   * [`u16`:`base`]
+   * [`bool`:`is_signed`]
+   * [`u16`:`nb_nonces`]
```

Comment:
super duper nit but worth considering whether we want to call this number of nonces or number of digits

Reply:
Given that you can have an extra nonce when `is_signed` is true I think `nb_nonces` make more sense.

Reply:
I agree with @Tibo-lg here, doing num digits can be misleading